### PR TITLE
Add application failover e2e

### PR DESCRIPTION
**Pull request by Poor12**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
Add application failover e2e

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
None

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Thanks, I will have a look later~

File: test/e2e/failover_test.go [hunk 1]
```diff
@@ -242,6 +242,234 @@ var _ = framework.SerialDescribe("failover testing", func() {
...
+								MaxGroups:     maxGroups,
+								MinGroups:     minGroups,
+							},
+						},
+					},
+					PropagateDeps: true,
+					Failover: &policyv1alpha1.FailoverBehavior{
+						Application: &policyv1alpha1.ApplicationFailoverBehavior{
+							DecisionConditions: policyv1alpha1.DecisionConditions{
+								TolerationSeconds: pointer.Int32(60),
```
Comment (XiShanYongYe-Chang) [inline test/e2e/failover_test.go:291]: Can this parameter be set to a short value? For E2E, the waiting time may be long.

Same as `GracePeriodSeconds` value.

File: test/e2e/failover_test.go [hunk 2]
```diff
@@ -242,6 +242,234 @@ var _ = framework.SerialDescribe("failover testing", func() {
...
+					for _, targetClusterName := range targetClusterNames {
+						// the target cluster should be overwritten to another available cluster
+						if !testhelper.IsExclude(targetClusterName, disabledClusters) {
+							return 0
+						}
+					}
+
+					return len(targetClusterNames)
+				}, pollTimeout, pollInterval).Should(gomega.Equal(minGroups))
+				framework.WaitDeploymentPresentOnClustersFitWith(disabledClusters, deploymentNamespace, deploymentName, func(deployment *appsv1.Deployment) bool { return true })
```
Comment (XiShanYongYe-Chang) [inline test/e2e/failover_test.go:463]: We can use the `ginkgo.By` module to organize the test separately.

Comment (Poor12) [reply test/e2e/failover_test.go:463]: Is it like:
```
ginkgo.By("xxxx", func(){
     framework.WaitDeploymentPresentOnClustersFitWith(disabledClusters, deploymentNamespace, deploymentName, func(deployment *appsv1.Deployment) bool {return true} )
})
```

File: test/e2e/failover_test.go (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply test/e2e/failover_test.go:291]: Is 10 seconds enough?

File: test/e2e/failover_test.go (hunk 2 above)
Comment (XiShanYongYe-Chang) [reply test/e2e/failover_test.go:463]: Yes, it seems better.

File: test/e2e/failover_test.go (hunk 1 above)
Comment (Poor12) [reply test/e2e/failover_test.go]: I feel like a little short. We need to wait the deployment running first. I feel like 30s is better.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**